elyra/airflow/bootstrapper.py

```diff
@@ -328,8 +328,19 @@ def execute(self) -> None:
                 f"executing python script using " f"'python3 {python_script}' to '{python_script_output}'"
             )
             t0 = time.time()
+
+            run_args = ["python3", python_script]
+
             with open(python_script_output, "w") as log_file:
-                subprocess.run(["python3", python_script], stdout=log_file, stderr=subprocess.STDOUT, check=True)
+                print("Processing file:", python_script)
+                try:
+                    result = subprocess.run(run_args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True)
+                    output = result.stdout.decode('utf-8')
+                    logger.info(f"Output: {output}")
+                    logger.info(f"Return code: {result.returncode}")
+                except subprocess.CalledProcessError as e:
+                    logger.error("Output: %s", e.output.decode("utf-8"))
+                    logger.error("Return code: %s", e.returncode)
```

Comment:
ah, the `raise`, of course :-) thank you. Yes, I just discussed this with a coworker in Airflow pipeline node green / red status, and we were like "oh wow, the more we catch explicitely, the less the status changes to red in airflow.
I was busy with other stuff, but promised him to look into it and you just gave the the way to trickle this up correctly to Airflow / KFP. Thanks a bunch, will change that.

Reply:
by the way, currently, NotebookFileOps / ipynb logs do not get written correctly. As I found out, they are only logged when base level log level of the system is DEBUG. I had to set log_output=True.

`papermill.execute_notebook(notebook, notebook_output, kernel_name=kernel_name, log_output=True, stdout_file=sys.stdout, stderr_file=sys.stderr)`

Will check whether raising of error happens correctly, too, there.

Reply:
> It looks like the pipeline node is green even on failure.

@harshad16 with your explicit raise of the CalledProcessError, you mean in the context of kubeflow pipelines, correct?

Cause I tested this on airflow and apparently, there, an explicit AirflowFailException is needed to get away from task status green ...

if one wants to fail the task without retries, one has to use AirflowFailException :-
Example :-

```
from airflow.exceptions import AirflowFailException
def task_to_fail():
    raise AirflowFailException("Our api key is bad!")
```

Reply:
>with your explicit raise of the CalledProcessError, you mean in the context of bootstrapper.py in kubeflow pipelines, correct?

yes explicitly with Kubeflow pipeline.
I haven't tested this with AirflowFailException.

Reply:
never mind the AirflowFailException for airflow ... I omitted the modified bootstrapper file in my airflow workbench images build, only had them in contrib snippet ... all good. That is, of course the raising of the exception also leads to task failure in Airflow 2.8.2.

Reply:
to get around log info:

`[2024-08-08, 20:14:37 UTC] {pod_manager.py:483} INFO - [base] AttributeError: 'CalledProcessError' object has no attribute 'return_code'
`

suggesting instead of

`raise subprocess.CalledProcessError(e.return_code, run_args)`

the following:

https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError

`raise subprocess.CalledProcessError(returncode=e.return_code, cmd=run_args)`

Reply:
no, that did not do the job, still in log in runtime, in this case airflow, related to the call:

```
[2024-08-09, 07:15:14 UTC] {pod_manager.py:466} INFO - [base]   File "/opt/app-root/src/jupyter-work-dir/bootstrapper.py", line 344, in execute
[2024-08-09, 07:15:14 UTC] {pod_manager.py:466} INFO - [base]     raise subprocess.CalledProcessError(returncode=e.return_code, cmd=run_args)
[2024-08-09, 07:15:15 UTC] {pod_manager.py:483} INFO - [base] AttributeError: 'CalledProcessError' object has no attribute 'return_code'
```
it is just at level info and might have more do do with subprocess library in python than anything else ...